src/plugin.js

```diff
@@ -33,182 +35,185 @@ const depFromErr = (err) => {
   return matches[1];
 };
 
-function NpmInstallPlugin(options) {
-  this.preCompiler = null;
-  this.compiler = null;
-  this.options = Object.assign(installer.defaultOptions, options);
-  this.resolving = {};
-
-  installer.checkBabel();
-}
-
-NpmInstallPlugin.prototype.apply = (compiler) => {
-  this.compiler = compiler;
+class NpmInstallPlugin {
+  constructor(options) {
+    this.preCompiler = null;
+    this.compiler = null;
+    this.options = Object.assign(installer.defaultOptions, options);
+    this.resolving = {};
```

Comment:
Potential memory leak, we should avoid keeping requests in memory

Reply:
I'm not quite sure what's the workaround should be, do you have any suggestions?

Reply:
request is the name of the module and not the request object btw

Reply:
anyway, this is existing code, we need improve tests still and code. Let's do it next?

Reply:
This we should solve here, can we avoid using `this.resolving`?

Reply:
can you suggest a workaround? Or we simply drop it? I think it was meant to prevent modules from firing multiple installs of the same package

Reply:
oh, I see, let's use `new Map` in this case and use `delete` when we do installation (with any error code)

Reply:
gotcha wip 👍

Reply:
or we can just use `new Set` instead since it's only storing booleans as values

Reply:
@alexander-akait fixed 👍